Remove clipped texts throughout the app when device font size is increased

osama256 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

This PR gives you the ability to increase or decrease the device font size without seeing clipped texts throughout the app.

the App behavior will be : 

1. when the device font size is decreased the App font size will decrease.
2. when the device font size is increased the App font size will increase.
3. when the device font size is increased and exceeded a certain larger level `PixelRatio.getFontScale() * default value > max value ? max value : default value * PixelRatio.getFontScale()` the App font size will not increase because any increasing above this level will show clipped texts throughout the app.
4. if the device font size is changed while the App is in the background, this change will not affect the App until close and reopen the App again.
5. this is similar to what YouTube does to handle the device font size change.


### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

 https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/7541

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Increase the device font size.
2. Open the App.
3. Verify there is no Text clipping throughout the app.

- [X] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [X] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [X] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [X] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] Any functional components have the `displayName` property
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose and it is
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn’t already exist
    - [X] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.


<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Increase the device font size.
2. Open the App.
3. Verify there is no Text clipping throughout the app.

- [X] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->

https://user-images.githubusercontent.com/51829206/200543093-9f61f174-af34-4ada-8cf9-6b9c2b2c6805.mov


#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

https://user-images.githubusercontent.com/51829206/200543186-161005f7-98aa-4953-8b0a-8f6cc5d605fa.mov

#### Desktop

https://user-images.githubusercontent.com/51829206/200543315-7c4af41f-df1b-417f-addd-2851013742ee.mov



<!-- Insert screenshots of your changes on the iOS platform-->





#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

https://user-images.githubusercontent.com/51829206/200543560-91e0c833-54d7-4e53-8045-ae4721fd5a5e.mov




#### Android
<!-- Insert screenshots of your changes on the Android platform-->


https://user-images.githubusercontent.com/51829206/169550493-72137556-94f3-4f6b-be31-a28de829a5a3.mov

osama256 committed:
Remove clipped texts throughout the app

osama256 committed:
Merge branch 'Expensify:main' into remove_clipped_texts

Santhosh-Sellavel commented:
Bug: 
The New Message notification badge is shown after changing font size

https://user-images.githubusercontent.com/85645967/183958324-0c3faea9-079a-4b5f-a383-7e394ee5c7ed.mov

Santhosh-Sellavel reviewed (changes requested):
Bug has been found while testing

osama256 committed:
Fix notification badge TranslateY

Signed-off-by: osama256 <51829206+osama256@users.noreply.github.com>

osama256 commented:
Fixed.
![Simulator Screen Shot - iPhone 13 - 2022-08-11 at 14 44 13](https://user-images.githubusercontent.com/51829206/184138064-5dae23bf-d313-4374-9ec5-38b370bf3305.png)


Also, there was another bug (badge icon alignment)

![before ](https://user-images.githubusercontent.com/51829206/184139206-0ce46e25-7681-4f8e-8813-f2668d77f56d.png)


Fixed here.

![after](https://user-images.githubusercontent.com/51829206/184139000-6c9264f0-3e49-4a0e-a5ec-1ccb42c47ff3.png)

Santhosh-Sellavel commented:
@osama256 Still the text `new message` is not centered.
<img width="1024" alt="Screenshot 2022-08-14 at 11 34 56 PM" src="https://user-images.githubusercontent.com/85645967/184549371-468e7898-6c3f-4e7d-84c5-7f5e865e8764.png">

Santhosh-Sellavel commented:
@osama256 bump

osama256 committed:
Fix new message vertical alignment

osama256 commented:
Fixed.
![android](https://user-images.githubusercontent.com/51829206/184863388-a47dd86d-8845-4a83-a5bd-ecdfd97831b8.png)

Santhosh-Sellavel commented on src/components/Button.js:
```diff
@@ -216,7 +216,7 @@ class Button extends Component {
                     </View>
                     {this.props.shouldShowRightIcon && (
-                        <View>
+                        <View style={styles.justifyContentCenter}>
```
@osama256 
Sorry, why this was change was made?
For whatever reason Isn't there any alternate solution,  as it could affect all button usage?

cc: @pecanoro

osama256 replied:
@Santhosh-Sellavel We made this change to fix one of [these bugs](https://github.com/Expensify/App/pull/10060#issuecomment-1193350895), `alignItemsCenter` style already used in the sibling `View` 
https://github.com/Expensify/App/blob/22a0cba7aaeaa786f04a63a5ad74c650c4c35001/src/components/Button.js#L207-L220
And since the parent `View` uses `flexRow` style, both children should use `alignItemsCenter` style, so I guess the developer forgot to add it.

Santhosh-Sellavel reviewed (commented):
@pecanoro 
This PR is almost done but seeking clarifications on a few things, then we should be good to merge this.

@osama256 Thanks for being patient, and helping this PR move forward. We're mostly done.
Can you answer those questions & pull the latest changes from `main`?

Thanks!

osama256 committed:
Merge branch 'Expensify:main' into remove_clipped_texts

Santhosh-Sellavel commented:
Seems conflicts again @osama256

osama256 committed:
fetch upstream

Santhosh-Sellavel commented:
@osama256 

Bug: Drop down arrow is not centered for larger text in iOS & Android
 
<img width="526" alt="Screenshot 2022-09-09 at 11 54 43 PM" src="https://user-images.githubusercontent.com/85645967/189418868-d864b0df-185c-4016-aa74-de3b952939b2.png">

Santhosh-Sellavel commented:
@pecanoro 

This is the PR that I've been testing for a longer period, every time I am ready to approve I am running into a new bug which is the reason this PR is tested for a longer period.

I requesting that this PR should be treated as a feature rather than a bug. I believe there will be no deploy blocker, but there is the slightest chance there may be a breakage which should be treated as new bugs.

Because this PR more or less has an impact on the whole app, which require more time as more eyes notice small bugs because these bugs are really easy to miss!

Once the last found bug [here](https://github.com/Expensify/App/pull/10312#issuecomment-1242325790) is fixed let's proceed with merging this one after testing a bit.

Please share your thoughts, thanks!

Santhosh-Sellavel reviewed (changes requested):
https://github.com/Expensify/App/pull/10312#issuecomment-1242325790

osama256 commented:
@Santhosh-Sellavel sorry for the delay, I'm working on it.

osama256 committed:
Fix picker icon alignment

osama256 commented:
Fixed.
![Screen Shot 2022-09-18 at 12 03 01 PM](https://user-images.githubusercontent.com/51829206/190896682-37b8f3c6-5a69-47dd-802b-cdde75346c6a.png)

Santhosh-Sellavel commented on src/styles/styles.js:
```diff
@@ -804,7 +804,7 @@ const styles = {
     picker: (disabled = false, error = false, focused = false) => ({
         iconContainer: {
-            top: variables.pickerIconTop,
+            top: Math.round(variables.inputHeight * 0.5) - 10,
```
This is weird, care to explain this?

osama256 replied:
TO vertically center a child in its parent we need padding top = half parent height - half child height, in our case half input height = 26 and half icon height = 10 so we need padding top = 16, but because the input height increases when the device font size increases so instead of using a hard coded value we can make it dynamic.

Santhosh-Sellavel replied:
This calculation, itself seems off can we move the computation variables at least?

osama256 replied:
Do you mean keeping `top: variables.pickerIconTop` in `styles.js` and in variables.js file using this `pickerIconTop: (getValueUsingPixelRatio(52, 72) * 0.5) - 10`?

pecanoro commented:
@Santhosh-Sellavel Sorry, I was on vacation last week. Yes, you are right, this PR has a lot of chances to bring some regressions due to the whole impact, I will bring it up to some of our most knowledgable React Native engineers so we can get an additional review.

pecanoro commented:
@Santhosh-Sellavel I discussed it with the team and if you think this PR is taking a lot of time to be reviewed so it doesn't bring any issues later, you should request a price increase, we love to aim for zero bugs/regressions even if it's a new feature. I feel you and it's true this new feature can be tricky so like I mentioned, I am bringing one of our engineers that is more experienced with React Native to do a final review, but still we should wait until this is ready to be merged for that final review.

Santhosh-Sellavel commented:
@pecanoro

Definitely, a lot of time is already invested in testing this one but the reason I've not requested a price increase can't figure out how this would go, so waiting till this gets merged.

> we love to aim for zero bugs/regressions even if it's a new feature.

Yeah, I agree with that too. We are doing our best here because of the affected volume, there is a slight we could miss small unnoticeable. 

> I am bringing one of our engineers that is more experienced with React Native to do a final review, but still, we should wait until this is ready to be merged for that final review.

Sounds good!

osama256 committed:
Merge branch 'Expensify:main' into remove_clipped_texts

Santhosh-Sellavel commented:
@osama256 Can you update the branch again with the latest main, thanks!

osama256 committed:
Merge branch 'Expensify:main' into remove_clipped_texts

osama256 commented:
Updated.

Santhosh-Sellavel commented:
Thanks @osama256

Santhosh-Sellavel reviewed (commented):
@osama256 Please merge with main. 
And update videos in the PR Description, Since there are so many changes from the last update, thanks!
Let's try to get this merged this week!

osama256 committed:
Merge branch 'Expensify:main' into remove_clipped_texts

osama256 commented:
@Santhosh-Sellavel The branch and screenshots have been updated.

Santhosh-Sellavel commented:
@pecanoro heads up this will be ready for final review hopefully tomorrow(as soon as the conflicts above are resolved, I'll quickly test and ping you again). As discussed [here](https://github.com/Expensify/App/pull/10312#issuecomment-1256183852) let's get someone from the internal to review this. I will add the recordings/PR Checklist followed by the review!

Santhosh-Sellavel commented:
@osama256 bump!

Santhosh-Sellavel commented:
@osama256 bump!

Santhosh-Sellavel commented:
@osama256 bump!

osama256 committed:
Fix conflict

osama256 committed:
Fetch upstream

osama256 committed:
Disallow picker font scaling

osama256 committed:
Merge branch 'remove_clipped_texts' of https://github.com/osama256/App into remove_clipped_texts

osama256 commented:
Updated.

Santhosh-Sellavel commented on src/components/TestToolRow.js:
```diff
@@ -14,7 +14,7 @@ const propTypes = {
 const TestToolRow = props => (
     <View style={[styles.flexRow, styles.mb6, styles.justifyContentBetween, styles.alignItemsCenter]}>
-        <View style={styles.flex3}>
+        <View style={styles.flex2}>
```
Why this?

Santhosh-Sellavel replied:
There are multiple issues there but I think you are referring to Preferences Page.

Santhosh-Sellavel replied:
But we updated the Button view style to fix that one, not sure how this fixed it. Can you briefly explain this thanks!

osama256 replied:
>Can you briefly explain this

Ok, this is what you get when using `styles.flex3` with a large font size.

![Simulator Screen Shot - iPhone 13 - 2022-11-21 at 20 40 04](https://user-images.githubusercontent.com/51829206/203134587-2d930f53-ba09-4cfa-baed-5dbd6afd9bef.png)


And this is what you get when using `styles.flex2` with a large font size.


![Simulator Screen Shot - iPhone 13 - 2022-11-21 at 20 40 13](https://user-images.githubusercontent.com/51829206/203134678-4d129813-f36c-4f2c-994a-9632baf5dac4.png)

As you can see, there is a difference between both, by using `styles.flex2` you can see the `Invalidate` word completely.

Santhosh-Sellavel commented:
@pecanoro Lets get someone for final review, before it doesn't run into any conflicts!

Santhosh-Sellavel commented:
Ah, conflicts again sorry @osama256 please resolve.

osama256 committed:
Merge branch 'main' into remove_clipped_texts

osama256 commented:
The conflict has been resolved.

Santhosh-Sellavel commented:
Thanks @osama256. @pecanoro lets get someone internal review this thanks!

pecanoro commented:
I am going to ask for a second review, CSS and styles is not my strength and I don't want to cause regressions.

pecanoro commented:
@Santhosh-Sellavel did you fill out the reviewer list?

Santhosh-Sellavel commented:
I didn't add it yet. Please go ahead with the request for the second review. I will fill it out once along with recordings once the second review is done as I mentioned [here](https://github.com/Expensify/App/pull/10312#issuecomment-1314212847) please excuse me on this thanks!

luacmartins commented on src/pages/home/report/FloatingMessageCounter/index.js:
```diff
@@ -25,7 +25,7 @@ const defaultProps = {
 };
 
-const MARKER_INACTIVE_TRANSLATE_Y = -30;
+const MARKER_INACTIVE_TRANSLATE_Y = -40;
```
Why did we change this value?

osama256 replied:
To completely hide the `FloatingMessageCounter` component when we are using a larger font size.
![Simulator Screen Shot - iPhone 13 - 2022-12-01 at 07 46 16](https://user-images.githubusercontent.com/51829206/204976224-db486e18-2f09-4511-9bcc-b6d74fe1fb81.png)

luacmartins commented on src/styles/variables.js:
```diff
@@ -1,6 +1,23 @@
+
+/**
+ * Calculate the fontSize, lineHeight and padding when the device font size is changed, In most cases users do not change their device font size so PixelRatio.getFontScale() = 1 and this
+ * method always return defaultValue (first param). when device font size increases/decrease, PixelRatio.getFontScale() value increases/decrease as well. this means if you hava a text
```
```suggestion
 * method always returns the defaultValue (first param). When the device font size increases/decreases, the PixelRatio.getFontScale() value increases/decreases as well. This means that if you have text
```

luacmartins commented on src/styles/variables.js:
```diff
@@ -1,6 +1,23 @@
+/**
+ * Calculate the fontSize, lineHeight and padding when the device font size is changed, In most cases users do not change their device font size so PixelRatio.getFontScale() = 1 and this
+ * method always return defaultValue (first param). when device font size increases/decrease, PixelRatio.getFontScale() value increases/decrease as well. this means if you hava a text
+ * and the 'fontSize' of it is 19, then the device font size changed to the 5th level on ios slider now the actual fontSize = 19 * PixelRatio.getFontScale() = 19 * 1.11 = 21.09 since we are
```
```suggestion
 * and its 'fontSize' is 19, the device font size changed to the 5th level on the iOS slider and the actual fontSize is 19 * PixelRatio.getFontScale() = 19 * 1.11 = 21.09. Since we are
```

luacmartins commented on src/styles/variables.js:
```diff
@@ -1,6 +1,23 @@
+ * Calculate the fontSize, lineHeight and padding when the device font size is changed, In most cases users do not change their device font size so PixelRatio.getFontScale() = 1 and this
+ * method always return defaultValue (first param). when device font size increases/decrease, PixelRatio.getFontScale() value increases/decrease as well. this means if you hava a text
+ * and the 'fontSize' of it is 19, then the device font size changed to the 5th level on ios slider now the actual fontSize = 19 * PixelRatio.getFontScale() = 19 * 1.11 = 21.09 since we are
+ * disallowing font scaling we need to calculate it manually, so we are using this equation : PixelRatio.getFontScale() * defaultValue > maxValue ? maxValue : defaultValue * PixelRatio.
```
```suggestion
 * disallowing font scaling we need to calculate it manually. We calculate it with: PixelRatio.getFontScale() * defaultValue > maxValue ? maxValue : defaultValue * PixelRatio
```

luacmartins commented on src/styles/variables.js:
```diff
@@ -1,6 +1,23 @@
+ * method always return defaultValue (first param). when device font size increases/decrease, PixelRatio.getFontScale() value increases/decrease as well. this means if you hava a text
+ * and the 'fontSize' of it is 19, then the device font size changed to the 5th level on ios slider now the actual fontSize = 19 * PixelRatio.getFontScale() = 19 * 1.11 = 21.09 since we are
+ * disallowing font scaling we need to calculate it manually, so we are using this equation : PixelRatio.getFontScale() * defaultValue > maxValue ? maxValue : defaultValue * PixelRatio.
+ * getFontScale() this equation means increases/decreases the fontSize when the device font size increases/decreases but do not increase it if the fontSize value exceed maxValue (second
```
```suggestion
 * getFontScale() This means that the fontSize is increased/decreased when the device font size changes up to maxValue (second param)
```

luacmartins commented on src/styles/variables.js:
```diff
@@ -1,6 +1,23 @@
+ * and the 'fontSize' of it is 19, then the device font size changed to the 5th level on ios slider now the actual fontSize = 19 * PixelRatio.getFontScale() = 19 * 1.11 = 21.09 since we are
+ * disallowing font scaling we need to calculate it manually, so we are using this equation : PixelRatio.getFontScale() * defaultValue > maxValue ? maxValue : defaultValue * PixelRatio.
+ * getFontScale() this equation means increases/decreases the fontSize when the device font size increases/decreases but do not increase it if the fontSize value exceed maxValue (second
+ * param).
```
```suggestion

```

luacmartins reviewed (changes requested):
Left a few comments. I still need to test this though.

osama256 committed:
Update src/styles/variables.js

Co-authored-by: Carlos Martins <luacmartins@gmail.com>

osama256 committed:
Update src/styles/variables.js

Co-authored-by: Carlos Martins <luacmartins@gmail.com>

osama256 committed:
Update src/styles/variables.js

Co-authored-by: Carlos Martins <luacmartins@gmail.com>

osama256 committed:
Update src/styles/variables.js

Co-authored-by: Carlos Martins <luacmartins@gmail.com>

osama256 committed:
Update src/styles/variables.js

Co-authored-by: Carlos Martins <luacmartins@gmail.com>

luacmartins reviewed (dismissed):
LGTM and tests well.

luacmartins commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

![web](https://user-images.githubusercontent.com/22219519/205147617-0dafa66f-a4c6-443b-bb3f-0f15222786aa.png)

</details>

<details>
<summary>Mobile Web - Chrome</summary>
![chrome](https://user-images.githubusercontent.com/22219519/205147674-8857b834-0c1f-4d09-bdce-6d345adab37d.png)

</details>

<details>
<summary>Mobile Web - Safari</summary>

![safari](https://user-images.githubusercontent.com/22219519/205147722-255e40e3-8150-4278-9a8c-65a8bce8fea8.png)

</details>

<details>
<summary>Desktop</summary>

<img width="1204" alt="desktop" src="https://user-images.githubusercontent.com/22219519/205147893-b23b8780-ed1d-4c88-83ec-b288a169c065.png">

</details>

<details>
<summary>iOS</summary>

![ios](https://user-images.githubusercontent.com/22219519/205147788-2d59d5d0-0ac2-41a1-a836-8e751ad1f355.png)

</details>

<details>
<summary>Android</summary>

![android](https://user-images.githubusercontent.com/22219519/205147818-aa837cd3-93e7-41bc-904e-9e8d4f0c04c4.png)

</details>

luacmartins commented:
@osama256 there's a lint warning about line length exceeding 190 characters. Can you please take a look?

osama256 committed:
Update variables.js

osama256 commented:
Updated.

luacmartins commented:
@osama256 we have conflicts now.

osama256 committed:
Merge branch 'main' into remove_clipped_texts

osama256 commented:
Resolved.

luacmartins reviewed (approved):
LGTM and tests well!

Santhosh-Sellavel commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

pecanoro commented on src/styles/styles.js:
```diff
@@ -29,7 +29,7 @@ const picker = {
     paddingBottom: 8,
     paddingTop: 23,
-    height: 50,
+    height: variables.inputHeight,
```
I can find any of the variables that you changed in this file. Am I going crazy?

pecanoro replied:
I think they might have been removed in one of those times when you merged with main

luacmartins replied:
I think they are added on this PR in `variables.js`, unless I'm missing something.

pecanoro replied:
Oh you are right, I was looking to a specific commit and couldn't  find them, like I said, going insane hehe

Santhosh-Sellavel commented:
I find these two bugs while testing now
### On Normal size, g is getting clipped here (I feel this could be not related to this but looks like well within scope too) 
#### Main
![Simulator Screen Shot - iPhone 14 - 2022-12-02 at 23 34 48](https://user-images.githubusercontent.com/85645967/205357313-8f624329-cdef-4b63-8b47-00a2fe66ff6a.png)

https://user-images.githubusercontent.com/85645967/205357321-b09b927b-5a7a-449a-b24b-749fb8424b30.mov


#### On this branch
![Simulator Screen Shot - iPhone 14 - 2022-12-02 at 23 35 21](https://user-images.githubusercontent.com/85645967/205357342-683d9b65-6807-41c1-b181-bf57f5d980a1.png)

https://user-images.githubusercontent.com/85645967/205357357-cafe68bf-58c9-4726-81fe-92cb736a3695.mov

### Search placeholder is not centered properly in larger sizes 
Okay this issue exists for normal font size too I will report it if not reported

Santhosh-Sellavel commented:
@pecanoro what should we do with the first issue, should we expect to solve it here?

As I see It's not related to what we are solving here, as the issue occurs only when the field has focus. So it might be irrelevant too, but I want to hear your thoughts too. Thanks!
cc: @luacmartins

pecanoro commented:
Let me retest that, does it happen in all platforms or only iOS?

Santhosh-Sellavel commented:
The first issue iOS only occurs on the main branch also for normal font size and the second issue is on all platforms I reported already in slack.

pecanoro commented:
> The first issue iOS only occurs on the main branch also for normal font size and the second issue is on all platforms I reported already in slack.

That happens even on the main branch, so I would say pretty unrelated. Also, I feel that is a regression of something else.

pecanoro commented:
Ok, I am making the decision of merging since those bugs were not introduced by this PR and it's out of scope, this one is handling an increase or decrease in size. @Santhosh-Sellavel If you can complete the list with videos it would be great even though @luacmartins already did it. If we wait longer, this PR will never get merged since we tend to get conflicts.

Santhosh-Sellavel commented:
I'll definitely do that, note story book is failing Rajat already reported in slack. Technically it would not have any effect, as this  PR mainly for Native devices

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3689736386) to production by @chiragsalian in version: 1.2.38-6 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

osama256 commented:
@parasharrajat Thanks for mentioning these issues here, but it seems these issues are not produced by this PR, and here is why.

1. for Picker height [inputHeight](https://github.com/Expensify/App/blob/b4d8aa459c4fe1dedc2327c8993f38e12ddb5d2b/src/styles/styles.js#L32) = `50` and it is the same value as in version `1.2.36` (two weeks ago) it was hardcoded `50`, I know it was `52` in the previous versions (months ago) like in version `1.1.38` but someone changed it later.

2. For the overlapping issue, we didn't do [this change](https://github.com/Expensify/App/commit/99971de55492d7acfe6f2792e3491e55707a26af#diff-8ab8ba45ab927f1430b8d1209d187e003a9cdfe476f809e4483ef64cdba253ebR737) and you can check it in [my repo](https://github.com/osama256/App/blob/184c49f82d0e06718802ebcec27bc9b710e0cb7c/src/styles/styles.js#L732)


6. For line height we have [lineHeightNormal](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/variables.js#L75) = `16` and [lineHeightLarge](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/variables.js#L76) = `18`, we used `lineHeightNormal` in [badgeText](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/styles.js#L628) and [peopleBadgeText](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/styles.js#L2479) and both were `16` in version `1.2.36` (two weeks ago)

![Screen Shot 2022-12-22 at 6 55 10 PM](https://user-images.githubusercontent.com/51829206/209210226-1524e7bc-de8f-4f58-b904-1e1a8579be7d.png)

![Screen Shot 2022-12-22 at 6 54 55 PM](https://user-images.githubusercontent.com/51829206/209210258-3fd9a503-a5fd-4cde-bc31-647ed52d840a.png)

And we used `lineHeightLarge` in [label](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/styles.js#L219), [textLabel](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/styles.js#L225), [mutedTextLabel](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/styles.js#L231), [formLabel](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/styles.js#L885) and [formHelp](https://github.com/Expensify/App/blob/70d87b0e6662c9fc06e85c0df35e3766fa4e1d04/src/styles/styles.js#L892) all of these were `18` in the previous version (1.2.36)

![Screen Shot 2022-12-22 at 6 55 33 PM](https://user-images.githubusercontent.com/51829206/209211344-43c19ed7-5989-4bb0-b1fc-88e20b9f26ae.png)

![Screen Shot 2022-12-22 at 6 56 04 PM](https://user-images.githubusercontent.com/51829206/209211414-09223266-2c83-479b-b986-cd7f2b1ebe7c.png)

NikkiWines commented:
👋 Hi all, we've identified this PR as the cause for https://github.com/Expensify/App/issues/13336. Since this modified styles for generic components, more detailed steps than "Verify there is no Text clipping throughout the app" would've been good to prevent bugs from slipping through.

Discussion on steps to be taken to ensure these bugs don't slip through can be found [here](https://expensify.slack.com/archives/C049HHMV9SM/p1671637209503109) cc: @mollfpr

osama256 commented:
>Discussion on steps to be taken to ensure these bugs don't slip through can be found [here](https://expensify.slack.com/archives/C049HHMV9SM/p1671637209503109)

 I can't open the link.
![Screen Shot 2022-12-26 at 4 44 54 PM](https://user-images.githubusercontent.com/51829206/209560190-59c1b98f-9208-4f05-9ccd-0609426209fd.png)

mollfpr commented:
@osama256 Have you join `#expensify-bugs` channel?

osama256 commented:
> @osama256 Have you join #expensify-bugs channel?

@mollfpr No, I have sent an email to join.

pecanoro commented:
@NikkiWines We discussed the clipped g here, it is not a regression of this PR, it was already on main before we merged the PR as @Santhosh-Sellavel pointed out [here](https://github.com/Expensify/App/pull/10312#issuecomment-1335638825).

pecanoro commented:
Though the emoji problem getting clipped on the top part could be coming from this one, but I haven't double check

osama256 commented:
>Though the emoji problem getting clipped on the top part could be coming from this one

@pecanoro Where is it?
